File: 백태균/week05/12945_피보나치_수.java
class Solution {
    public int solution(int n) {
        // 피보나치 점화식: f(n) = f(n-1) + f(n-2)

        int[] fibo = new int[n+1];

        fibo[0] = 0;  // 0번째를 0으로 지정
        fibo[1] = 1;  // 1번째를 1로 지정

        int i = 2;
        while (i <= n) {
            // f(47)부터는 21억을 넘기 때문에 int 범위를 초과함
            fibo[i] = (fibo[i-1] + fibo[i-2]) % 1234567;  // 그래서 fibo(n) 구한 값을 바로 % 1234567 해서 나머지 값을 저장
            i++;
        }

        return fibo[n];
    }
}

File: 김진효/week05/12945_피보나치_수.py
# [성능 요약] 메모리: 455 MB 시간: 327.37 ms 
def solution(n):
    f = [0] * (n+1)
    f[0] = 0
    f[1] = 1

    for i in range(2,n+1):
        f[i] = f[i-2] + f[i-1]
    
    return f[n] % 1234567

print(solution(3))


# 런타임 에러
def not_solution(n):
    def fibo(n):
        if n>=2 and memo[n] ==0:
            memo[n] = fibo(n-2) + fibo(n-1) 
        return memo[n]

    memo = [0] * (n+1)
    memo[0] = 0
    memo[1] = 1

    return fibo(n) % 1234567